### PR TITLE
Explore by tags page

### PR DESCRIPTION
**Pull request by ghost**
This is work that @jaronheard started in #669. I've moved it into a separate PR so that @chrislupdx  can pick it up!

### CODE REVIEW

File: packages/2019-template/src/components/TemplateAPICard/index.js [hunk 1]
```diff
@@ -11,13 +11,7 @@ import api from "../../state/template-api-data/api";
...
   useEffect(() => {
     init();
-  }, [
-    /*
-    https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects
-
-    Add second argument to prevent useEffect running init() again
-    */
-  ]);
+  }, [init]);
```
Comment (Dianna) [inline packages/2019-template/src/components/TemplateAPICard/index.js]: This line needs to be reverted. You may will probably need to add an eslint-disable-next-line. Written like this, there are cases where init can be run again and we don't want that behavior

File: packages/2019-template/src/components/TemplateAPICard/templateAPICardMeta.js [hunk 2]
```diff
@@ -31,9 +31,7 @@ const TemplateAPICardMeta = (/* data */) => ({
   ),
   shareText:
     "Ice cream candy canes brownie marzipan jelly chocolate bar marshmallow.",
-  tags: [
-    /* "Transportation", "Bus", "Rail", "Portland" */
-  ],
+  tags: ["Transportation", "Bus", "Rail", "Portland", "Stuff"],
```
Comment (Dianna) [inline packages/2019-template/src/components/TemplateAPICard/templateAPICardMeta.js]: This needs to stay commented out. We don't want cards that are a WIP to show up on the cards to go see.

File: packages/component-library/stories/CardList.story.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      slug: "template-file-card",
+      project: "@hackoregon/2019-template",
+      component: SampleCard
+    },
+    {
+      slug: "demo-card",
+      project: "@hackoregon/2019-template",
+      component: SampleCardB
+    }
+  ]
```
Comment (Dianna) [inline packages/component-library/stories/CardList.story.js]: I see how you're using these two cards for proof of concept. @jaronheard can have these cards tagged or not? I'll have my disaster project cards merged in tomorrow and we could use those

File: packages/component-library/src/CardList/CardList.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+            {drawer}
+          </Drawer>
+        </Hidden>
+      </nav>
+      <main className={classes.content}>
+        <h1>Did you know?</h1>
+        <div>
+          <ul className={classes.entriesList}>
+            {entries
+              .filter(entry => checkTags(entry.component.tags, activeTags))
```
Comment (Dianna) [inline packages/component-library/src/CardList/CardList.js]: The behavior currently is that if you select 2 tags (say transportation and transit), then the cards are _expanded_ to include cards with _either_ tag. I'm not sure if that is the desired behavior or if _further narrowing down_ is the desired behavior. The latter feels more natural to me. @jaronheard, any thoughts?

File: packages/2018/src/components/CardList.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
```
Comment (jaronheard) [inline packages/2018/src/components/CardList.js]: This file is no longer used and can be deleted

File: packages/component-library/src/CardList/CardList.js (hunk 4 above)
Comment (jaronheard) [reply packages/component-library/src/CardList/CardList.js]: Hmm, I'm not totally sure. I think I'm going to throw this out in #team-design and see if anyone else has thoughts on this when it's one long list.

However, once you add categories, as shown in our Figma, then I think I have a clear idea of what is supposed to happen:
Select first tag in a category => Narrow selection
Select additional tags in a category => Expand selection

Comment (Dianna) [reply packages/component-library/src/CardList/CardList.js]: That is the behavior here. So... it's as expected then? The dynamic still seems weird to me but I'm not strongly attached to it working one way or the other

**Comment by cosmolightfoot** [comment]
I added Material-UI's nested list component to the filter section. I have concerns about the UX of reverse filtering -- since all checkboxes are checked by default, the user looking for something specific has to uncheck everything that isn't that thing rather than checking what they are looking for. @jaronheard would love your thoughts on this.

**Comment by cosmolightfoot** [comment]
Also when the filters list is fully expanded you can't quite scroll to the bottom to see the last filters.

File: packages/component-library/src/CardList/CardList.js [hunk 6]
```diff
@@ -92,38 +107,157 @@ const checkTags = (entryTags, activeTags) => {
...
-              label={tag}
-              onChange={() =>
-                setActiveTags({ ...activeTags, [tag]: !activeTags[tag] })
-              }
-            />
-          </li>
-        ))}
-      </ul>
+      <List
+        component="nav"
```
Comment (Dianna) [inline packages/component-library/src/CardList/CardList.js]: I'm not sure what this should be, but probably not `nav`?

**Comment by Dianna** [comment]
I've had a chance to review this more thoroughly now. There are a couple of areas for improvement but we're almost there :)

File: packages/component-library/src/CardList/CardList.js (hunk 4 above)
Comment (jaronheard) [reply packages/component-library/src/CardList/CardList.js]: Examples:
_Selected Tags (category):_ What shows up
_Transportation (topic):_ Transportation
_Transportation (topic),  Housing (topic):_ Transportation OR Housing
_Transportation (topic), Bar Chart (visualization):_ Transportation AND Bar Chart
_Transportation (topic), Housing (topic), Bar Chart (visualization):_ (Transportation OR Housing)  AND Bar Chart
_Transportation (topic), Bar Chart (visualization), Portland (location):_ Transportation AND Bar Chart AND Portland

File: packages/2018/src/tags.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+export default {
+  topics: ["Transportation", "Disaster Resilience", "Housing"],
+  locations: ["Portland", "Nationwide", "Your City"],
+  visualizations: ["Bar Chart", "Cloropleth Map", "Scatterplot"]
+};
```
Comment (Dianna) [inline packages/2018/src/tags.js]: yay! We really needed a list

File: packages/2019-housing/src/components/AduDistributions/index.js [hunk 8]
```diff
@@ -8,13 +8,10 @@ import aduDistributionsMeta from "./aduDistributionsMeta";
...
-    () => {
-      init();
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    },
-    [init]
-  );
+  useEffect(() => {
+    init();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [init]);
```
Comment (Dianna) [inline packages/2019-housing/src/components/AduDistributions/index.js]: Remove init from the brackets

File: packages/2019-housing/src/components/BlackPopulationChange/index.js [hunk 9]
```diff
@@ -11,13 +11,7 @@ import api from "../../state/black-population-change/api";
...
   useEffect(() => {
     init();
-  }, [
-    /*
-    https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects
-
-    Add second argument to prevent useEffect running init() again
-    */
-  ]);
+  }, [init]);
```
Comment (Dianna) [inline packages/2019-housing/src/components/BlackPopulationChange/index.js]: Remove init from the brackets

File: packages/2019-housing/src/components/HolcRedlining/index.js [hunk 10]
```diff
@@ -8,13 +8,10 @@ import holcRedliningMeta from "./holcRedliningMeta";
...
-    () => {
-      init();
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    },
-    [init]
-  );
+  useEffect(() => {
+    init();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [init]);
```
Comment (Dianna) [inline packages/2019-housing/src/components/HolcRedlining/index.js]: Remove init from the brackets

File: packages/2019-housing/src/components/HomeAppreciation/index.js [hunk 11]
```diff
@@ -8,13 +8,10 @@ import homeAppreciationMeta from "./homeAppreciationMeta";
...
-    () => {
-      init();
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    },
-    [init]
-  );
+  useEffect(() => {
+    init();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [init]);
```
Comment (Dianna) [inline packages/2019-housing/src/components/HomeAppreciation/index.js]: Remove init from brackets

File: packages/2019-housing/src/components/HouseholdIncomeByRace/index.js [hunk 12]
```diff
@@ -8,13 +8,10 @@ import householdIncomeByRaceMeta from "./householdIncomeByRaceMeta";
...
-    () => {
-      init();
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    },
-    [init]
-  );
+  useEffect(() => {
+    init();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [init]);
```
Comment (Dianna) [inline packages/2019-housing/src/components/HouseholdIncomeByRace/index.js]: Remove init from brackets

File: packages/2019-housing/src/components/HousingDisplacement/index.js [hunk 13]
```diff
@@ -11,13 +11,7 @@ import api from "../../state/housing-displacement/api";
...
   useEffect(() => {
     init();
-  }, [
-    /*
-    https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects
-
-    Add second argument to prevent useEffect running init() again
-    */
-  ]);
+  }, [init]);
```
Comment (Dianna) [inline packages/2019-housing/src/components/HousingDisplacement/index.js]: Remove init from brackets

File: packages/2019-transportation/src/components/DisturbanceStops/index.js [hunk 14]
```diff
@@ -8,13 +8,10 @@ import disturbanceStopsMeta from "./disturbanceStopsMeta";
...
-    () => {
-      init();
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    },
-    [init]
-  );
+  useEffect(() => {
+    init();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [init]);
```
Comment (Dianna) [inline packages/2019-transportation/src/components/DisturbanceStops/index.js]: Remove init from brackets

File: packages/component-library/src/CardList/CardList.js [hunk 15]
```diff
@@ -77,260 +91,216 @@ const useStyles = makeStyles(theme => ({
...
-                      ...activeTags,
-                      [other]: !activeTags[other]
-                    })
-                  }
-                />
-              </ListItem>
-            ))}
-          </List>
-        </Collapse>
+        {Object.keys(tagsList).map(category => (
```
Comment (Dianna) [inline packages/component-library/src/CardList/CardList.js]: Nice job with the map!

File: packages/component-library/src/CardList/CardList.js [hunk 16]
```diff
@@ -77,260 +91,216 @@ const useStyles = makeStyles(theme => ({
...
+                className={classes.categoryListText}
+                primary={category.replace(/^\w/, c => c.toUpperCase())}
+              />
+              {categoryOpeners[category] ? <ExpandLess /> : <ExpandMore />}
+            </ListItem>
+            <Collapse
+              in={categoryOpeners[category]}
+              timeout="auto"
+              unmountOnExit
+            >
```
Comment (Dianna) [inline packages/component-library/src/CardList/CardList.js]: The caret is facing the opposite of what it should be for opened / closed

File: packages/component-library/src/CardList/CardList.js [hunk 17]
```diff
@@ -77,260 +91,216 @@ const useStyles = makeStyles(theme => ({
...
-        <div>
+    <ThemeProvider theme={MaterialTheme}>
+      <div className={classes.root}>
+        <CssBaseline />
+        <AppBar position="fixed" className={classes.appBar}>
+          <Header title="Story Cards" />
+        </AppBar>
+        <nav className={classes.drawer} aria-label="mailbox folders">
+          <Hidden smUp implementation="css">
+            <Drawer
```
Comment (Dianna) [inline packages/component-library/src/CardList/CardList.js]: The Drawer should be behind the AppBar
Screen Shot 2019-09-08 at 5 06 41 PM

File: packages/component-library/src/CardList/CardList.js (hunk 16 above)
Comment (Dianna) [reply packages/component-library/src/CardList/CardList.js]: Actually, it should be facing right when closed and down when open

File: packages/component-library/src/CardList/CardList.js (hunk 17 above)
Comment (Dianna) [reply packages/component-library/src/CardList/CardList.js]: Never mind, we'll need to redo this anyways

**Review by ghost — COMMENTED** [review]
Amazing. But I can't approve this PR since I started it haha

**Review by jaronheard — APPROVED** [review]
🚢 when ✅

**Pull request merged**